packages/fxa-auth-server/lib/payments/paypal.ts

```diff
@@ -346,6 +346,36 @@ export class PayPalHelper {
     }));
   }
 
+  /**
+   * Removes Paypal billing agreements on a customer if they paid with
+   * Paypal but no longer have an active/past_due subscription.
+   */
+  async conditionallyRemoveBillingAgreement(
+    customer: Stripe.Customer
+  ): Promise<boolean> {
+    const billingAgreementId = this.stripeHelper.getCustomerPaypalAgreement(
+      customer
+    );
+    if (!billingAgreementId) {
+      return false;
+    }
+    const paypalSubscription = customer.subscriptions?.data.find(
+      (sub) =>
+        ['active', 'past_due'].includes(sub.status) &&
+        sub.collection_method === 'send_invoice'
+    );
```

Comment:
deja vu all over again :-P

Reply:
Indeed, it'll be nice to have a single function that does this in shared that can be used everywhere.